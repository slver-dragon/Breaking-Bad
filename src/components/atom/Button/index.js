import React from "react";
import style from "./style.module.scss";

const Button = ({ img, text }) => {
  const imageElement = img !== "" ? <img src={img} alt="" /> : "";
  const textElement = text !== "" ? <span>{text}</span> : "";
  return (
    <div className={style.button}>
      {imageElement}
      {textElement}
    </div>
  );
};

export default Button;
