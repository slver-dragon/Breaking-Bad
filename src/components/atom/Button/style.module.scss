@import "../../../assets/styles/style";

.button {
  display: flex;
  align-items: center;
  text-decoration: none;
  cursor: pointer;
  img {
    margin-right: 0.75rem;
  }
  span {
    @include mediumFont;
    color: $black;
    margin-left: 0.75rem;
  }
}
