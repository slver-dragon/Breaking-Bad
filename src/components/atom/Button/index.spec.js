import React from "react";
import { shallow } from "enzyme";
import Button from "../Button";

describe("Button atom testing:", () => {
  let component = null;
  const setComponent = (props) => shallow(<Button {...props}/>);
  
  test("is render full", () => {
    component = setComponent({img:'./test.jpg', text:'Test message'});
    expect(component).toMatchSnapshot();
  });

  test("is render empty", () => {
    component = setComponent({img:'', text:''});
    expect(component).toMatchSnapshot();
  });
});
