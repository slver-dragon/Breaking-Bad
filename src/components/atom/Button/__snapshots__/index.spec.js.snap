// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button atom testing: is render empty 1`] = `
<button
  className="button"
/>
`;

exports[`Button atom testing: is render full 1`] = `
<button
  className="button"
>
  <img
    alt=""
    src="./test.jpg"
  />
  <span>
    Test message
  </span>
</button>
`;
